js/coins.js

```diff
@@ -0,0 +1,222 @@
+
+let globalTimeOut = null
+
+
+  const payload = {
+      coin: 'https://api.coingecko.com/api/v3/coins',
+      fCoinPrice: 'https://min-api.cryptocompare.com/data/pricemulti?fsyms',
+      sCoinPrice: 'tsyms=USD'
+  }
+
+   const state = {mark:[] , }
+
+   let list = []
+
+  $(document).ready(init) 
+
+   function init(){
+    changeBodyCoins()
+       $("#about").on("click" , changeBodyAbout)
+       $("#realTime").on("click" , changeBodyRealTime)
+       $("#coins").on("click" , changeBodyCoins)
+  }
+ 
+
+  function clearSearchInput(){
+    console.log("kj")
+      $("#searchInput").val("")
+  }
+
+  async function drawAllCoins(){
+      $("#content").html(drawLoader())
+      const coins = await getCoinsList()
+      if(!coins) return
+      $("#content").html("")
+      coins.map( (c) => {
+         list.push(c) 
+         drawCoin(c)
+      })
+}
+
+
+  async function getCoinsList(){
+
+      try{
+        const result = await fetch(payload.coin)
+        const jsonR = await result.json()
+        return jsonR
+      }
+      catch(er){
+          console.log(er)
+      }
+  }
+
+  function checkboxF(value){
+    if (!isString(value)) return
+    clearTimeout(globalTimeOut)
+    globalTimeOut = setTimeout(()=>{
+      alertToMuchTime()
+      changeBodyRealTime()
+    },120000)
```

Comment:
what is this timeout? for caching? you should provide here much more rubust logic, using the timestamp the coin information cached/